### DISCUSSION TITLE
如何出块前计算基础奖励？

### DISCUSSION THREAD

**Discussion by ruyichang** [nervosnetwork/ckb#2408] [General]
我想在get_block_template的同时拿到一个类似btc基础奖励的一个值，比如12

请问通过epoch可以计算出这个值吗？

如果可以的话请问小数是多少位？

这个值和浏览器里面【基础奖励】怎么换算

**Comment by doitian** [comment]
Epoch 的奖励在一段时间内是固定的，初始是 [1_917_808_21917808 ](https://github.com/nervosnetwork/ckb/blob/a8dbc63c4e1a1a6c4432979cb48e8df831560ef5/spec/src/consensus.rs#L31) Shannons，每过 8760 个 Epoch 减半一次（无法整除向下取整）。所以 0~8759 都是 1_917_808_21917808 Shannons 的基础奖励。

Epoch 的奖励再平分给 epoch 中的每个块，除不尽的部分平分给前 N 个块。参考 https://github.com/nervosnetwork/ckb/wiki/Memo

所有 CKB 的计算都是使用的 Shannon 为最小单位，Shannon 不可再分割，不能表示成小数。1CKB = 10^8 Shannons
